src/Codegen/Builders/TypeBuilder.hack

```diff
@@ -32,12 +32,18 @@ abstract class TypeBuilder<T as \Slack\GraphQL\__Private\GraphQLTypeInfo> {
     }
 
     public function build(HackCodegenFactory $cg): CodegenClass {
-        return $cg
+        $class = $cg
             ->codegenClass($this->getGeneratedClassName())
             ->setIsFinal()
             ->setExtendsf('\%s', static::SUPERCLASS)
             ->addConstant($this->generateNameConstant($cg))
             ->addTypeConstant($this->generateTypeConstant($cg));
+
+        if ($this is ITypeWithFieldsBuilder) {
```

Comment:
It might be nicer to use a trait for this - to avoid putting this in the base class. Seems a bit odd for the base class to need to know about this.